azurerm/internal/services/network/tests/resource_arm_virtual_hub_connection_test.go

```diff
@@ -0,0 +1,272 @@
+			}
+		}
+
+		if !found {
+			return fmt.Errorf("Connection %q was not found", name)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMVirtualHubConnectionDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Network.VirtualHubClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		if rs.Type != "azurerm_virtual_hub_connection" {
+			continue
+		}
+
+		virtualHubId := rs.Primary.Attributes["virtual_hub_id"]
+		id, err := networkSvc.ParseVirtualHubID(virtualHubId)
+		if err != nil {
+			return err
+		}
+
+		resourceGroup := id.ResourceGroup
+		hubName := id.Name
+		name := rs.Primary.Attributes["name"]
+
+		resp, err := client.Get(ctx, resourceGroup, hubName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on network.VirtualHubClient: %+v", err)
+			}
+		}
+
+		// since it's been deleted, that's fine
+		if resp.VirtualHubProperties == nil {
+			return nil
+		}
+		props := *resp.VirtualHubProperties
+		if props.VirtualNetworkConnections == nil {
+			return nil
+		}
+
+		conns := *props.VirtualNetworkConnections
+		for _, conn := range conns {
+			if conn.Name != nil && *conn.Name == name {
+				return fmt.Errorf("Connection %q still exists", name)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMVirtualHubConnection_basic(data acceptance.TestData) string {
+	template := testAccAzureRMVirtualHubConnection_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_virtual_hub_connection" "test" {
+  name                      = "acctestvhubconn-%d"
+  virtual_hub_id            = azurerm_virtual_hub.test.id
+  remote_virtual_network_id = azurerm_virtual_network.test.id
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMVirtualHubConnection_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMVirtualHubConnection_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_virtual_hub_connection" "import" {
+  name                      = azurerm_virtual_hub_connection.test.name
+  virtual_hub_id            = azurerm_virtual_hub_connection.test.virtual_hub_id
+  remote_virtual_network_id = azurerm_virtual_hub_connection.test.remote_virtual_network_id
+}
+`, template)
+}
+
+func testAccAzureRMVirtualHubConnection_complete(data acceptance.TestData) string {
+	template := testAccAzureRMVirtualHubConnection_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_virtual_hub_connection" "test" {
+  name                                       = "acctestvhubconn-%d"
+  virtual_hub_id                             = azurerm_virtual_hub.test.id
+  remote_virtual_network_id                  = azurerm_virtual_network.test.id
+  allow_hub_to_remote_vnet_transit           = true
+  allow_remote_vnet_to_use_hub_vnet_gateways = false
+  enable_internet_security                   = true
+}
+`, template, data.RandomInteger)
```

Comment:
could we define two virtual hubs for the complete test, and then have another update test where we go basic -> compelte -> basic to ensure we correctly handel multiple?

Reply:
updated